orig 719 cand 723 -42 +46 _ZN18pump_amm_recovered12instructions4swap8trade_v212execute_sel
@@ -281 +281 @@
-ja +0xe
+ja +0x11
@@ -295 +295,4 @@
-ja +0x1ac
+ldxdw r2, [r1 + 0x8]
+add64 r2, -0x1
+stxdw [r1 + 0x8], r2
+ja -0x7c
@@ -325 +328 @@
-ja -0xf3
+ja -0xf6
@@ -359 +362 @@
-ja -0xbc
+ja -0xbf
@@ -369 +372 @@
-ja -0xc6
+ja -0xc9
@@ -386 +389 @@
-jne r9, 0x2, +0x132
+jne r9, 0x2, +0x138
@@ -402 +405 @@
-ja +0x124
+ja +0x12a
@@ -414 +417 @@
-ja -0x14f
+ja -0x152
@@ -428,0 +432 @@
+ldxdw r8, [r10 - 0x598]
@@ -431,3 +435,7 @@
-ja +0x109
-ldxdw r9, [r7 + 0x8]
-ldxdw r8, [r7 + 0x60]
+ja -0x108
+mov64 r1, 0x2
+mov64 r2, r8
+call X
+ldxdw r1, [r7 + 0x8]
+stxdw [r10 - 0x678], r1
+ldxdw r9, [r7 + 0x60]
@@ -437,0 +446,4 @@
+mov64 r8, r9
+add64 r8, 0x8
+mov64 r2, r10
+add64 r2, -0x320
@@ -439,4 +450,0 @@
-add64 r1, 0x8
-mov64 r2, r10
-add64 r2, -0x320
-stxdw [r10 - 0x678], r1
@@ -447,2 +455,3 @@
-jne r0, 0x0, +0xc
-ldxdw r1, [r9 + 0x0]
+jne r0, 0x0, +0xd
+ldxdw r1, [r10 - 0x678]
+ldxdw r1, [r1 + 0x0]
@@ -450,2 +459 @@
-ldxdw r3, [r10 - 0x678]
-stxdw [r3 + 0x18], r2
+stxdw [r8 + 0x18], r2
@@ -453 +461 @@
-stxdw [r3 + 0x10], r2
+stxdw [r8 + 0x10], r2
@@ -455 +463 @@
-stxdw [r3 + 0x8], r2
+stxdw [r8 + 0x8], r2
@@ -457,3 +465,4 @@
-stxdw [r3 + 0x0], r1
-ldxdw r8, [r7 + 0x60]
-ldxdw r9, [r7 + 0x8]
+stxdw [r8 + 0x0], r1
+ldxdw r1, [r7 + 0x8]
+stxdw [r10 - 0x678], r1
+ldxdw r9, [r7 + 0x60]
@@ -462 +471,2 @@
-ldxdw r4, [r9 + 0x0]
+ldxdw r2, [r10 - 0x678]
+ldxdw r4, [r2 + 0x0]
@@ -496,0 +507,2 @@
+add64 r9, 0x8
+stxdw [r10 - 0x150], r9
@@ -498,2 +509,0 @@
-add64 r8, 0x8
-stxdw [r10 - 0x150], r8
@@ -508,2 +518 @@
-jeq r9, 0x2, +0x1
-ja +0xbd
+jne r9, 0x2, +0xbb
@@ -549 +558 @@
-jeq r1, 0x2, -0x17a
+jeq r1, 0x2, -0x183
@@ -669 +678 @@
-ja +0x1a
+ja +0x18
@@ -687 +696 @@
-ja -0x209
+ja -0x212
@@ -689 +698 @@
-ja -0x20b
+ja -0x214
@@ -691,3 +700 @@
-ja -0x20d
-ldxdw r8, [r10 - 0x598]
-ja -0x20f
+ja -0x216
@@ -695 +702 @@
-ja -0x211
+ja -0x218
@@ -715,5 +722,2 @@
-jne r2, 0x0, -0x225
-ldxdw r2, [r1 + 0x8]
-add64 r2, -0x1
-stxdw [r1 + 0x8], r2
-ja -0x229
+jne r2, 0x0, -0x22c
+ja -0x1b5
